source/Mission.cpp

```diff
@@ -1228,40 +1246,22 @@ Mission Mission::Instantiate(const PlayerInfo &player, const shared_ptr<Ship> &b
 
 	// If cargo is being carried, see if we are supposed to replace a generic
 	// cargo name with something more specific.
-	if(!cargo.empty())
+	if(cargoObjective.CanBeRealized())
 	{
-		const Trade::Commodity *commodity = nullptr;
-		if(cargo == "random")
-			commodity = PickCommodity(*sourceSystem, *result.destination->GetSystem());
-		else
-		{
-			for(const Trade::Commodity &option : GameData::Commodities())
-				if(option.name == cargo)
-				{
-					commodity = &option;
-					break;
-				}
-			for(const Trade::Commodity &option : GameData::SpecialCommodities())
-				if(option.name == cargo)
-				{
-					commodity = &option;
-					break;
-				}
-		}
-		if(commodity)
-			result.cargo = commodity->items[Random::Int(commodity->items.size())];
-		else
-			result.cargo = cargo;
+		result.cargo = cargoObjective.RealizeCargo(*sourceSystem, *result.destination->GetSystem());
+		result.cargoSize = cargoObjective.RealizeCount();
 	}
-	// Pick a random cargo amount, if requested.
-	if(cargoSize || cargoLimit)
+	// If outfit is needed, see if we are supposed to replace a generic
```

Comment:
Based on a few lines above for the cargo check where there's a two-line comment, you forgot to put the second line in this comment. The missing line should be: '// outfit name with something more specific.'